dali/ft/filecopy.cpp

```diff
@@ -2623,6 +2623,49 @@ void FileSprayer::setTarget(IDistributedFile * target)
     }
 }
 
+void FileSprayer::checkTargetPath(RemoteFilename & filename)
+{
+    StringBuffer targetFilePath;
+
+    filename.getRemotePath(targetFilePath);
+    // prune the leading "//" from remote path
+    targetFilePath.remove(0, 2);
+
+    StringBuffer netaddress;
+    filename.queryIP().getIpText(netaddress);
+    const char * pnetaddress = netaddress.str();
+
+    if (streq("127.0.0.1", pnetaddress))
+        throwError1(DFTERR_LocalhostAddressUsed, targetFilePath.str());
+
+    //remove ip address from the beginning of the path
+    targetFilePath.remove(0, netaddress.length());
+#ifdef _DEBUG
+    LOG(MCdebugInfo, unknownJob, "Target file path is '%s'", targetFilePath.str());
+#endif
+
+    const char * ptargetFilePath = targetFilePath.str();
+    const char pathSep = filename.getPathSeparator();
+    const char dotString[]    = {pathSep, '.', pathSep, '\0'};
+    const char dotDotString[] = {pathSep, '.', '.', pathSep, '\0'};
+
+    const char * isDotString = strstr(ptargetFilePath, dotString);
+    const char * isDotDotString = strstr(ptargetFilePath, dotDotString);
+    if ((isDotDotString != nullptr) || (isDotString != nullptr))
+        throwError3(DFTERR_InvalidTargetPath, ptargetFilePath, dotDotString, dotString);
```

Comment:
Strictly speaking, it would not be a security violation if relative path like:
```
./somedir/someotherdir/../someotherdir2/
```

were used. In other words, as long as it's within the spray LZ directrory, however it got there via relative qualifiers, it's still ok.
NB: could also be via soft/hard links setups. If so, the below code wouldn't find it.

I'm not sure this is particularly important, but..